File: python/jobs.py

import splunklib.client as client
import utils, sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
try: 
    if len(sys.argv) == 1:
        print "syntax: jobs.py [list | show jobid | results jobid | delete jobid]"
        sys.exit()
    
    service = client.connect(**opts.kwargs)

    if sys.argv[1] == "list":
        #LAB: print each job's ID and search string (hint: eventSearch) to STDOUT
        
    elif sys.argv[1] == "show":
        job = #LAB: get the job named in sys.argv[2]
        #LAB: print all job attributes to STDOUT

    elif sys.argv[1] == "results":
        job = #LAB: get the job named in sys.argv[2]
        if not job.is_done():
            raise Exception("Job not complete")
        response = #LAB: get the job results
        print response
        
    elif sys.argv[1] == "delete":
        job = #LAB: get the job named in sys.argv[2]
        #LAB: cancel the job
        print "  job %s deleted" % job.sid
 
    else:
        raise Exception("unknown command \"%s\"" % sys.argv[1])

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))

File: java/jobs.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class jobs {

    public static void main(String[] args) {

		try {
			if(args.length < 1) {
				System.out.println("syntax: jobs.py [list | show jobid | results jobid | delete jobid]");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);
			if(args[0].equalsIgnoreCase("list")) {
				//LAB: print each job's ID and search string (hint: eventSearch) to STDOUT
			}
			else if(args[0].equalsIgnoreCase("show")) {
				Job job = //LAB: get the job named in args[1]
				//LAB: print all job attributes to STDOUT
			}
			else if(args[0].equalsIgnoreCase("results")) {
				Job job = //LAB: get the job named in args[1]
		        InputStream response = //LAB: get the job results
		        int chr = 0;
				while (chr != -1) System.out.write(chr = response.read());
			}
			else if(args[0].equalsIgnoreCase("delete")) {
				Job job = //LAB: get the job named in args[1]
				//LAB: cancel the job
				System.out.println("  " + args[1] + " deleted.");
			}
			else {
				throw new Exception("Unknown command \"" + args[0] + "\"");
			}
			
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}